check.go

```diff
@@ -101,9 +102,9 @@ func (s *Site) checkContentPage(page *contentPageCheckData) (problems []string)
 			}
 
 			if node.Type == blackfriday.Link {
-				// Require that relative paths link to the actual .md file, so that browsing
-				// docs on the file system works.
-				if isPathOnly && u.Path != "" && !strings.HasSuffix(u.Path, ".md") {
+				// Require that relative paths link to the actual .md file, i.e not the the "foo" folder in the case of
+				// of "foo/index.md", so that browsing docs on the file system works.
+				if isPathOnly && u.Path != "" && filepath.Ext(u.Path) == "" {
```

Comment:
Do we have unit tests for these checks or not? 

I think `filepath.Ext(u.Path) == ""` should be `filepath.Ext(u.Path) != ""`

Reply:
@vrto that would break the predicate, what we want to catch here are links to "foo" which works because the page is an "index.md". So we only want to focus on paths where we don't have any extension at all.

Reply:
Ah, this was building a problem case, sorry I misread. All good!